cli/pcluster/utils.py

```diff
@@ -393,19 +394,41 @@ def verify_stack_creation(stack_name, cfn_client):
         LOGGER.debug(resource_status)
     if status != "CREATE_COMPLETE":
         LOGGER.critical("\nCluster creation failed.  Failed events:")
-        events = get_stack_events(stack_name, raise_on_error=True)
-        for event in events:
-            if event.get("ResourceStatus") == "CREATE_FAILED":
-                LOGGER.info(
-                    "  - %s %s %s",
-                    event.get("ResourceType"),
-                    event.get("LogicalResourceId"),
-                    event.get("ResourceStatusReason"),
-                )
+        _log_stack_failure_recursive(stack_name, indent=2)
         return False
     return True
 
 
+def _log_stack_failure_recursive(stack_name, indent):
+    """Log stack failures in recursive manner, until there is no substack layer."""
+    events = get_stack_events(stack_name, raise_on_error=True)
+    for event in events:
+        if event.get("ResourceStatus") == "CREATE_FAILED":
+            _log_failed_cfn_event(event, indent=indent)
+            if event.get("ResourceType") == "AWS::CloudFormation::Stack":
+                # Sample substack error:
+                # "Embedded stack arn:aws:cloudformation:us-east-2:704743599507:stack/
+                # parallelcluster-fsx-fail-FSXSubstack-65ITLJEZJ0DQ/
+                # 3a4ecf00-51e7-11ea-8e3e-022fd555c652 was not successfully created:
+                # The following resource(s) failed to create: [FileSystem]."
+                substack_capture = re.search(
+                    ".+/(parallelcluster-.+)/".format(stack_name), event.get("ResourceStatusReason")
```

Comment:
You're using `format` but there are no placeholders to replace in the string.
Then we should use the `PCLUSTER_STACK_PREFIX` variable from the constants.py.